Gems/AWSGameLift/Code/AWSGameLiftClient/Source/Activity/AWSGameLiftCreateSessionActivity.cpp

```diff
@@ -63,15 +70,21 @@ namespace AWSGameLift
             return request;
         }
 
-        AZStd::string CreateSession(
-            const Aws::GameLift::GameLiftClient& gameliftClient,
-            const AWSGameLiftCreateSessionRequest& createSessionRequest)
+        AZStd::string CreateSession(const AWSGameLiftCreateSessionRequest& createSessionRequest)
         {
+            AZStd::string result = "";
+
+            auto gameliftClient = AZ::Interface<IAWSGameLiftInternalRequests>::Get()->GetGameLiftClient();
```

Comment:
DRY: For another PR, this same code is in a lot of places (same code block to get and validate client). IMHO we should reduce bloat like this.

Reply:
I am not sure how to dedupe further here, and what is the value of doing it.
```
AZStd::shared_ptr<GameLiftClient> GetGameLiftClient(const AZStd::string& logTag)
{
    if (m_gameliftClient) {
        return m_gameliftClient;
    } else {
        AZ_Error(logTag, false, AWSGameLiftClientMissingErrorMessage);
        return nullptr;
    }
}

// caller
...
AZStd::string result = "";
auto gameliftClient = AZ::Interface<IAWSGameLiftInternalRequests>::Get()->GetGameLiftClient();
if (!gameliftClient)
{
    return result;
}
...
```
With above thing, you just dedupe the `AZ_Error` here. And most importantly, you make assumption here that client getter function should always log error when it is nullptr. 
You can optimize more, like pass a lambda function as parameter to this getter function so caller has power to control the log, but the weirdest thing is why do we even need a logging lambda function as client getter function parameter.